woocommerce/woocommerce-android
# Fix/dashboard custom range tracking

Closes: #13055

### Description

In Android we are tracking some of the custom range dashboard interaction using a legacy event that should've been removed back when we worked on the dynamic dashboard project. The event is `dashboard_stats_custom_range_tab_selected` and was used back when the dashboard had tabs to select the date ranges. 
We track this event every time the user clicks on the drop-down and selectect `custom`. Note the first time the user selects `custom` the event `dashboard_stats_custom_range_add_button_tapped` will be fired. But after adding a custom range subsequent taps on `custom range` option will trigger the legacy event. 

Additionally, I renamed the function `onTabSelected` to `onRangeChanged` to improve readability and get rid of the old `tab` concept. 

### Testing information

1. Add `DashboardStatsCard` (aka performance) to the dashboard.
2. Select `custom` range option and set a custom range
3. Tap again on `custom` option from the drop down. 
4. Check that for either step 3 and 4 the event `woocommerceandroid_dashboard_stats_custom_range_add_button_tapped` is tracked.

Repeat the above steps for `TopPerfomersCard` and check tracking is correctly fired. 

### The tests that have been performed

The above. 

### Images/gif



## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
